File: core/data/src/main/java/com/kisslink/data/repository/ITransferRepository.java
package com.kisslink.data.repository;

import androidx.lifecycle.LiveData;

import com.kisslink.domain.TransferRecord;

import java.util.List;

/**
 * 傳輸紀錄倉庫介面——使用領域模型 {@link TransferRecord} 作為 API 邊界。
 *
 * <p>由 {@link LiveDataTransferRepository} 實作（Room + Mapper）。
 * 上層（ViewModel）依賴本介面，可在測試中替換為 fake 實作。
 */
public interface ITransferRepository {

    LiveData<List<TransferRecord>> getAllRecords();

    LiveData<List<TransferRecord>> getByBatch(long batchId);

    /**
     * 搜尋：在 fileName/peerDeviceName 中模糊匹配。
     *
     * @param direction "SEND"、"RECEIVE"，或 null/空字串（不過濾）。
     */
    LiveData<List<TransferRecord>> search(String query, String direction);

    void insert(TransferRecord record);

    void delete(long id);

    void deleteByBatch(long batchId);

    void clearAll();
}

File: core/domain/src/main/java/com/kisslink/domain/repository/ITransferRepository.kt
package com.kisslink.domain.repository

import com.kisslink.domain.model.TransferRecord
import kotlinx.coroutines.flow.Flow

/** Domain contract for transfer history persistence. */
interface ITransferRepository {
    fun getAllRecords(): Flow<List<TransferRecord>>
    fun getRecentRecords(limit: Int): Flow<List<TransferRecord>>
    fun getByBatch(batchId: Long): Flow<List<TransferRecord>>

    suspend fun insert(record: TransferRecord): Long
    suspend fun delete(id: Long)
    suspend fun deleteByBatch(batchId: Long)
    suspend fun clearAll()
}
